pkg/controller/metrics/exporters/lvm_exporter/lvm_exporter.go

```diff
@@ -0,0 +1,150 @@
+			"Shows ReadThroughput",
+			volumeLabel, nil,
+		),
+		ResponseTime: prometheus.NewDesc("OpensSDS_Volume_ResponseTime_ms",
+			"Shows ReadThroughput",
+			volumeLabel, nil,
+		),
+		ServiceTime: prometheus.NewDesc("OpensSDS_Volume_ServiceTime_ms",
+			"Shows ServiceTime",
+			volumeLabel, nil,
+		),
+		Utilization: prometheus.NewDesc("OpensSDS_Volume_Utilization_prcnt",
+			"Shows Utilization in percentage",
+			volumeLabel, nil,
+		),
+	}
+
+}
+
+//Describe function.
+//It essentially writes all descriptors to the prometheus desc channel.
+func (collector *lvmCollector) Describe(ch chan<- *prometheus.Desc) {
+
+	//Update this section with the each metric you create for a given collector
+	ch <- collector.IOPS
+	ch <- collector.ReadThroughput
+	ch <- collector.WriteThroughput
+	ch <- collector.ResponseTime
+	ch <- collector.ServiceTime
+	ch <- collector.Utilization
+}
+
+//Collect implements required collect function for all promehteus collectors
+func (collector *lvmCollector) Collect(ch chan<- prometheus.Metric) {
+
+	collector.mu.Lock()
+	defer collector.mu.Unlock()
+
+	//Implement logic here to determine proper metric value to return to prometheus
+	//for each descriptor
+	metricList := []string{"IOPS", "ReadThroughput", "WriteThroughput", "ResponseTime", "ServiceTime", "UtilizationPercentage"}
+	//Todo : Need to read list from a config file
+	volumeList := []string{"sda", "loop0"}
+	metricDriver := lvm.MetricDriver{}
+	metricDriver.Setup()
+	for _, volume := range volumeList {
+		metricArray, _ := metricDriver.CollectMetrics(metricList, volume)
+		fmt.Println(metricArray)
+		for _, metric := range metricArray {
+			instanceLabel := metric.InstanceID
+			//unitLabel := "Unit:"+metric.Unit
+			switch metric.Name {
+			case "IOPS":
+				ch <- prometheus.MustNewConstMetric(collector.IOPS, prometheus.GaugeValue, metric.MetricValues[0].Value, instanceLabel)
+			case "ReadThroughput":
+				ch <- prometheus.MustNewConstMetric(collector.ReadThroughput, prometheus.GaugeValue, metric.MetricValues[0].Value, instanceLabel)
+			case "WriteThroughput":
+				ch <- prometheus.MustNewConstMetric(collector.WriteThroughput, prometheus.GaugeValue, metric.MetricValues[0].Value, instanceLabel)
+			case "ResponseTime":
+				ch <- prometheus.MustNewConstMetric(collector.ResponseTime, prometheus.GaugeValue, metric.MetricValues[0].Value, instanceLabel)
+			case "ServiceTime":
+				ch <- prometheus.MustNewConstMetric(collector.ServiceTime, prometheus.GaugeValue, metric.MetricValues[0].Value, instanceLabel)
+
+			case "UtilizationPercentage":
+				ch <- prometheus.MustNewConstMetric(collector.Utilization, prometheus.GaugeValue, metric.MetricValues[0].Value, instanceLabel)
+
+			}
+		}
+	}
+
+}
+func validateCliArg(arg1 string) string {
+	num, err := strconv.Atoi(arg1)
+	if (err != nil) || (num > 65535) {
+
+		fmt.Println("Please enter a valid port number")
+		os.Exit(1)
+	}
+	return arg1
+}
+func main() {
```

Comment:
I'm afraid I don't understand why `main` method is defined here.

Reply:
lvm_exporter is an independent process which user can start if required.